circle.yml

```diff
@@ -88,6 +89,55 @@ step-library:
         command: ccache --show-stats
 
 
+  - &save-node_modules-cache
+      save_cache:
+        name: Save node_modules cache
+        key: 'node_modules/v1/{{ arch }}/{{ checksum "package.json" }}'
+        paths: [ "node_modules" ]
+  - &restore-node_modules-cache
+      restore_cache:
+        name: Restore node_modules cache
+        keys:
+          - 'node_modules/v1/{{ arch }}/{{ checksum "package.json" }}'
+          - 'node_modules/v1/{{ arch }}'
+  - &save-mason_packages-cache
+      save_cache:
+        name: Save mason_packages cache
+        key: 'mason_packages/v1/{{ arch }}/{{ checksum "cmake/mason-dependencies.cmake" }}'
+        paths: [ "mason_packages/.binaries" ]
+  - &restore-mason_packages-cache
+      restore_cache:
+        name: Restore mason_packages cache
+        keys:
+          - 'mason_packages/v1/{{ arch }}/{{ checksum "cmake/mason-dependencies.cmake" }}'
+          - 'mason_packages/v1/{{ arch }}'
+  - &save-ccache
+      save_cache:
+        name: Save ccache
+        key: 'ccache/v1/{{ arch }}/{{ .Environment.CIRCLE_JOB }}/{{ checksum ".circle-branch" }}/{{ checksum ".circle-sha1" }}'
+        paths: [ "~/.ccache" ]
+  - &restore-ccache
+      restore_cache:
+        name: Restore ccache
+        keys:
+          - 'ccache/v1/{{ arch }}/{{ .Environment.CIRCLE_JOB }}/{{ checksum ".circle-branch" }}/{{ checksum ".circle-sha1" }}'
```

Comment:
I don't think so. However, unless builds complete out of order, the next fallback is likely to find the desired cache, because when a key prefix finds multiple matches, the most recently written one wins.

Reply:
Hm, but that prefix marching also applies to partial SHA1 strings?

Reply:
Not sure what you mean. There's no way these keys could match part of a SHA1. If the first one doesn't match anything exactly, the second one is tried. That will match any cache for the same arch/job/branch combination, and if there are two or more matches, the most recent one will be selected.